sycl/source/detail/graph_impl.cpp

```diff
@@ -572,8 +577,10 @@ bool modifiable_command_graph::end_recording(
 
 executable_command_graph::executable_command_graph(
     const std::shared_ptr<detail::graph_impl> &Graph, const sycl::context &Ctx)
-    : MTag(rand()),
-      impl(std::make_shared<detail::exec_graph_impl>(Ctx, Graph)) {
+    : impl(std::make_shared<detail::exec_graph_impl>(Ctx, Graph)) {
+  std::default_random_engine randEngine;
+  std::uniform_int_distribution<int> dist(0, RAND_MAX);
+  MTag = dist(randEngine);
```

Comment:
`MTag` is currently unused by the implementation, I think a better fix here might be to just remove the member variable and avoiding making the code more verbose with the explicit rand initalization.

Reply:
+1, unless we plan to give `MTag` any use in the future.